Realm/RLMQueryUtil.mm

```diff
@@ -71,6 +71,14 @@ static void RLMPrecondition(bool condition, NSString *name, NSString *format, ..
     return prop;
 }
 
+NSPredicate *RLMValidatedPredicate(id predicateFormat, va_list args) {
+    if (predicateFormat && ![predicateFormat isKindOfClass:[NSString class]]) {
```

Comment:
What's the point of this check? The methods are all declared as taking a NSString and we don't check for violations of the static type anywhere else in the api.